src/control/cmd/dmg/pretty/storage_test.go

```diff
@@ -172,6 +172,176 @@ PCI:%s Model:%s FW:%s Socket:%d Capacity:%s
 	}
 }
 
+func TestPretty_PrintNVMetaMap(t *testing.T) {
+	var (
+		controllerA = storage.MockNvmeController(1)
+		controllerB = storage.MockNvmeController(2)
+		controllerC = storage.MockNvmeController(1)
+		controllerD = storage.MockNvmeController(2)
+		controllerE = storage.MockNvmeController(1)
+		controllerF = storage.MockNvmeController(2)
+	)
+	controllerA.SmdDevices = nil
+	controllerB.SmdDevices = nil
+	controllerE.SmdDevices = []*storage.SmdDevice{
+		{
+			UUID:      common.MockUUID(0),
+			TargetIDs: []int32{0, 1, 2},
+			Rank:      0,
+			State:     "NORMAL",
+		},
+		{
+			UUID:      common.MockUUID(1),
+			TargetIDs: []int32{3, 4, 5},
+			Rank:      0,
+			State:     "FAULTY",
+		},
+	}
+	controllerF.SmdDevices = []*storage.SmdDevice{
+		{
+			UUID:      common.MockUUID(2),
+			TargetIDs: []int32{6, 7, 8},
+			Rank:      1,
+			State:     "NORMAL",
+		},
+		{
+			UUID:      common.MockUUID(3),
+			TargetIDs: []int32{9, 10, 11},
+			Rank:      1,
+			State:     "FAULTY",
+		},
+	}
+	for name, tc := range map[string]struct {
+		hsm         control.HostStorageMap
+		expPrintStr string
+	}{
+		"no controllers": {
+			hsm: mockHostStorageMap(t, &mockHostStorage{"host1", &control.HostStorage{}}),
+			expPrintStr: `
+-----
+host1
+-----
+  No NVMe devices detected
+`,
+		},
+		"no smd devices on controllers": {
+			hsm: mockHostStorageMap(t,
+				&mockHostStorage{
+					"host1",
+					&control.HostStorage{
+						NvmeDevices: storage.NvmeControllers{
+							controllerA,
+							controllerB,
+						},
+					},
+				},
+			),
+			expPrintStr: fmt.Sprintf(`
+-----
+host1
+-----
+PCI:%s Model:%s FW:%s Socket:%d Capacity:%s
+  No SMD devices found
+
+PCI:%s Model:%s FW:%s Socket:%d Capacity:%s
+  No SMD devices found
+
+`,
+				controllerA.PciAddr, controllerA.Model, controllerA.FwRev,
+				controllerA.SocketID, humanize.Bytes(controllerA.Capacity()),
+				controllerB.PciAddr, controllerB.Model, controllerB.FwRev,
+				controllerB.SocketID, humanize.Bytes(controllerB.Capacity())),
+		},
+		"single smd device on each controller": {
+			hsm: mockHostStorageMap(t,
+				&mockHostStorage{
+					"host1",
+					&control.HostStorage{
+						NvmeDevices: storage.NvmeControllers{
+							controllerC,
+							controllerD,
```

Comment:
```suggestion
							controlled,
```

Reply:
If you add controlled to the list of ignored words here you won't get this error.  In fact the code is currently clean so you probably should, or somebody else will have to do it in future.

https://github.com/daos-stack/daos/blob/master/ci/patch_src_in_place#L4